### PR TITLE
[langchain_community] Update/Insert texts to the Milvus vectorstore

### PR DESCRIPTION
**Pull request by rgupta2508**
Update/Insert texts to the Milvus vectorstore.

        Inserting data when the collection is already exists
        and need to insert data based on collection schema,
        Metadata keys will need to be present for all inserted values. At
        the moment there is no None equivalent in Milvus.
        This is based on upsert data based on add_text functionality

- [ ] **PR title**: "package: description"
  - Where "package" is whichever of langchain, community, core, experimental, etc. is being modified. Use "docs: ..." for purely docs changes, "templates: ..." for template changes, "infra: ..." for CI changes.
  - Example: "community: add foobar LLM"

- [ ] **PR message**: ***Delete this entire checklist*** and replace with
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change
    - **Twitter handle:** if your PR gets announced, and you'd like a mention, we'll gladly shout you out!

- [ ] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [ ] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, hwchase17.

### CODE REVIEW

**Comment by rgupta2508** [comment]
@hwchase17 Please review, Adding function to upsert text directly based on one of the field from metadata

**Comment by baskaryan** [comment]
cc @xiaofan-luan

**Comment by rgupta2508** [comment]
@xiaofan-luan Pleas review

**Comment by rgupta2508** [comment]
@baskaryanPleas review if this can we merged

cc:- @xiaofan-luan/

Thanks

File: libs/community/langchain_community/vectorstores/milvus.py [hunk 1]
```diff
@@ -1057,3 +1057,61 @@ def upsert(
...
+        if metadatas is None or len(metadatas) == 0:
+            return self.add_texts(texts=texts, metadatas=metadatas, **kwargs)
+        else:
+            values = [metadata[field_name] for metadata in metadatas]
+        # Convert to set for optimize expression
+        values = list(set(values))
+        if values is not None and len(values):
+            try:
+                if field_name == "ids":
+                    self.delete(ids=values)
```
Comment (xiaofan-luan) [inline libs/community/langchain_community/vectorstores/milvus.py:1105]: why do we not use upsert interface directly inside pymilvus but delete then insert?

Comment (xiaofan-luan) [reply libs/community/langchain_community/vectorstores/milvus.py:1105]: if we first delete then insert there seems to be no consistency guarantee

**Comment by rgupta2508** [comment]
@xiaofan-luan  Can you please suggest in this case, 

Actually current upsert interface only works with IDs, what if some one want to delete some of the row and adding n ew row in-place of those rows.
Basically embedding of one file can be added to multiple rows, based on splitter. now just want to update embedding for the same file with updated content , so this time it may create more or less row based on content and splitter.

Please review this or can you please suggest any other another possible solution.

**Comment by xiaofan-luan** [comment]
so far the best you can do is to query with expressions to find. Then upsert with primary keys.

We could think of add upsert with expression in the future

**Comment by rgupta2508** [comment]
Ok thank you for your quick response.
But i cant get key also, no of key(row) may not be same for one expression, it depends on metadata .
Fo example for, one expression i got 3 rows, but i need to add 5 row in place of those embedding row as some of the document might be splitted to multiple chunks . and how can will handle those chunks.?

I just tried to extend this method with expression :- https://github.com/langchain-ai/langchain/blob/c769421aa4fc4783968e069d800ef85191ce87da/libs/community/langchain_community/vectorstores/milvus.py#L1037
One option i can think is, deleting data using pymilvus interface, and adding using langchain.

**Comment by xiaofan-luan** [comment]
> Ok thank you for your quick response @xiaofan-luan . But i cant get key also, no of key(row) may not be same for one expression, it depends on metadata . Fo example for, one expression i got 3 rows, but i need to add 5 row in place of those embedding row as some of the document might be splitted to multiple chunks . and how can will handle those chunks.?
> 
> I just tried to extend this method with expression :-
> 
> https://github.com/langchain-ai/langchain/blob/c769421aa4fc4783968e069d800ef85191ce87da/libs/community/langchain_community/vectorstores/milvus.py#L1037
> 
> 
> One option i can think is, deleting data using pymilvus interface, and adding using langchain.

upsert can be used as insert as well. If no exsiting entity, you can simply add a new row while upsert

**Pull request closed without merging**